src/v/kafka/client/producer.h

```diff
@@ -48,8 +48,9 @@ class producer {
     produce(model::topic_partition tp, model::record_batch&& batch);
 
     ss::future<> stop() {
+        _as.request_abort();
         return ssx::parallel_transform(
-          std::move(_partitions),
+          _partitions,
```

Comment:
Right but the difference is within `producer::send` within the last callback `in_flight` will be true. When `in_flight` is true we don't want to deallocate the partitions because they hold promises for unresolved futures.